Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -634,20 +655,25 @@ WarpXOpenPMDPlot::DumpToFile (ParticleContainer* pc,
   //
   // define positions & offsets
   //
-  SetupPos(currSpecies, counter.GetTotalNumParticles(), charge, mass);
-  SetupRealProperties(currSpecies, write_real_comp, real_comp_names, write_int_comp, int_comp_names, counter.GetTotalNumParticles());
-
+  const unsigned long long NewParticleVectorSize = counter.GetTotalNumParticles() + ParticleFlushOffset;
+  m_ParticleSetUp = false;
+  if (counter.GetTotalNumParticles() > 0 and ParticleFlushOffset == 0) {
+      // This will trigger meta-data flush for particles the first-time non-zero number of particles are flushed.
+      m_ParticleSetUp = true;
+  }
+  SetupPos(currSpecies, NewParticleVectorSize, charge, mass, isBTD);
+  SetupRealProperties(currSpecies, write_real_comp, real_comp_names, write_int_comp, int_comp_names, NewParticleVectorSize, isBTD);
   // open files from all processors, in case some will not contribute below
   m_Series->flush();
-
   for (auto currentLevel = 0; currentLevel <= pc->finestLevel(); currentLevel++)
     {
       uint64_t offset = static_cast<uint64_t>( counter.m_ParticleOffsetAtRank[currentLevel] );
-
+      // For BTD, the offset include the number of particles already flushed
+      if (isBTD) offset += ParticleFlushOffset;
       for (ParticleIter pti(*pc, currentLevel); pti.isValid(); ++pti) {
          auto const numParticleOnTile = pti.numParticles();
          uint64_t const numParticleOnTile64 = static_cast<uint64_t>( numParticleOnTile );
-
+         if (numParticleOnTile == 0) continue;
```

Comment:
Did we need this permanently?
I think we need to add a comment here, because this works for regular diags with zero particles in a species.
Maybe we also want to limit this `continue` to BTD for now.

Reply:
I remember needing it -- but let me check again. otherwise we can add `if (isBTD)` conditionality here

Reply:
```suggestion
         // Do not call storeChunk() with zero-sized particle tiles:
         //   https://github.com/openPMD/openPMD-api/issues/1147
         if (numParticleOnTile == 0) continue;
```